### PR TITLE
Batch Creation and Deletion of Waypoints

### LINKED ISSUE #5: Allow batch creation and deletion of Waypoint's

### PR DESCRIPTION
**Pull request by jimgeng**
Batch creation and deletion of waypoints now exposed as `POST` and `DEL` endpoints for the `/waypoints` route, with corresponding updated swagger documentation.

Preliminary testing (in Insomnia) has been done to ensure basic functionality, further edge-case tests probably should be done to ensure code strength.

Other small changes:
- capitalization of response messages is more consistent.
- changed server start address to include the `localhost` keyword (without this my firewall acts up, no functional difference with or without otherwise).

### CODE REVIEW

File: controllers/waypoint_controller.go [hunk 1]
```diff
@@ -213,11 +261,72 @@ func DeleteWaypoint(c echo.Context) error {
...
+	for i := 0; i < len(waypointids.IDs); i++ {
+		// ID is represented as string in json, which gets converted to integer here after binding.
+		curID, convErr := strconv.Atoi(waypointids.IDs[i])
+		if convErr != nil {
+			return c.JSON(http.StatusBadRequest, responses.ErrorResponse{
+				Message: "Invalid IDs",
+				Data:    convErr.Error()})
+		}
+		println(curID)
+		dbAction := db.Delete(&models.Waypoint{}, curID)
```
Comment (21chanas3) [inline controllers/waypoint_controller.go]: Ideally, we should check that all of the IDs actually do exist before running any delete operation to avoid partial deletion

File: models/waypoint_model.go [hunk 2]
```diff
@@ -28,3 +28,15 @@ type Waypoint struct {
...
+type WaypointIDs struct {
+	/*
+		IDs is an array of strings because idk how to implement
+		custom unmarshalling of the json strings into ints.
+		I have chosen to simply convert the strings to integers
+		in the route function in waypoint_controller.go (see DeleteWaypointBatch())s
+		Please lmk if there is a way to do this in the unmarshalling process
+		instead of the actual route.
+	*/
+	IDs []string `json:"ids" example:"1,2,3"`
```
Comment (21chanas3) [inline models/waypoint_model.go]: As per your comment creating a struct here is not necessary. A good resource is this: https://pkg.go.dev/encoding/json?utm_source=godoc#Unmarshal

The JSON format should just be a simple JSON integer array, i.e. [1, 2, 3, 4]

File: main.go [hunk 3]
```diff
@@ -53,5 +56,5 @@ func main() {
 	//Websockets
 	e.Any("/socket.io/", controllers.WebsocketHandler())
 
-	e.Logger.Fatal(e.Start(":1323"))
+	e.Logger.Fatal(e.Start("localhost:1323"))
```
Comment (21chanas3) [inline main.go:59]: Did you encounter errors without "localhost"?

**Review by 21chanas3 — CHANGES_REQUESTED** [review]
Change batch deletion behavior and eliminate intermediate struct used in JSON parsing

File: main.go (hunk 3 above)
Comment (jimgeng) [reply main.go:59]: Windows pops up with a firewall whitelist dialog, running `go run main.go` without `localhost` keyword runs a temp executable in the windows temp folder, and the firewall pops up with a whitelist dialog without localhost.
image

**Review by 21chanas3 — APPROVED** [review]
lgtm
